### PR TITLE
Mainenance: adapted solution to be compatible with VS2017 and VSC. Updated dependencies. Switching from * .xproj+project.json to *.csproj

### PR DESCRIPTION
**Pull request by kloned**
Did some maintenance job to make solution compatible with Visual Studio 2017 and Visual Studio Code.
- No changes in source code;
- Updated dependencies;
- Switched from `*.xproj` and `project.json` to `*.csproj`
- Updated `global.json`
- Tests: 21 pass (RealProxy + DispatchProxy).
`LogInfoWithProperStackTrace_WhenTask()` sometimes fails, however. Could be due the fact that exception get thrown asynchronously in `AsyncMethodWithException` and another task is attached to it in `LoggingAdvice`'s Invoke (through `resultTask.ContinueWith`). In this case, as I understand, not only the "main" task should be awaited explicitly, but its follower should be awaited too.
Simply put: when I put `await` at the end of `resultTask.ContinueWith` body definition - the problem seems to be gone.

### CODE REVIEW

**Comment by ValeraT1982** [comment]
Thanks a lot. I appreciate your help.
